libsolidity/codegen/ir/IRGenerator.cpp

```diff
@@ -802,29 +802,24 @@ void IRGenerator::generateImplicitConstructors(ContractDefinition const& _contra
 string IRGenerator::deployCode(ContractDefinition const& _contract)
 {
 	Whiskers t(R"X(
-		<#loadImmutables>
-			let <var> := mload(<memoryOffset>)
-		</loadImmutables>
-
-		codecopy(0, dataoffset("<object>"), datasize("<object>"))
-
-		<#storeImmutables>
-			setimmutable(0, "<immutableName>", <var>)
-		</storeImmutables>
-
-		return(0, datasize("<object>"))
+		let <codeOffset> := <allocateUnbounded>()
+		codecopy(<codeOffset>, dataoffset("<object>"), datasize("<object>"))
+		<#immutables>
+			setimmutable(<codeOffset>, "<immutableName>", <value>)
+		</immutables>
+		return(<codeOffset>, datasize("<object>"))
 	)X");
+	t("allocateUnbounded", m_utils.allocateUnboundedFunction());
+	t("codeOffset", m_context.newYulVariable());
```

Comment:
We need to allocate a variable here because the above is not enclosed in a block (its own scope)? If so, could we enclose it in a scope to avoid the variable allocator (and get it sorted by disambiguator later)?

Probably helps in no way (apart from simplifying the code here), just trying to understand.

Reply:
Enclosing it in a block only helps half way:
```
let a := 1
{
  let a := 2
}
```
will still lead to an error

Reply:
It is not that likely to get a clash with variable names, but if it happens, it's very annoying I would say.